compiler/dyno/lib/resolution/resolution-types.cpp

```diff
@@ -152,10 +153,14 @@ const UntypedFnSignature* UntypedFnSignature::get(Context* context,
 CallInfo::CallInfo(const uast::FnCall* call) {
   // set the name (simple cases only)
   if (auto called = call->calledExpression()) {
-    if (auto id = called->toIdentifier())
+    if (auto id = called->toIdentifier()) {
       name_ = id->name();
+      // determine if this is an operator call based on its name
+      isOpCall_ = uast::isOpName(name_);
```

Comment:
I think not; would it make sense to swap this out for an assert that the name is not an op name?

Reply:
I think that would be fine, but maybe I shouldn't be the only one making that decision 😄

Reply:
I'll be the other one then 😄  (as well as anyone else who stops by here while I'm waiting for paratest)
I think it's pretty reasonable since we're assuming that FnCall can't be an operator; this encodes that assumption and if it becomes false in the future, it will be very obvious to change this.

Reply:
I think the case to be careful with is that the compiler-generated calls will still need to have their `isOpCall` determined by the name or we have to add some additional argument to the code creating them. In other words, by the time we are calling `resolveGeneratedCall`, we should have the "is it an operator" part figured out, one way or another.

Reply:
But anyway I would agree that `CallInfo(const uast::FnCall* call)` shouldn't be able to generate something with `isOpCall` being true, for this specific line.